videostreamer.c

```diff
@@ -323,7 +323,11 @@ int
 vs_write_packet(const struct VSInput * const input,
 		struct VSOutput * const output, AVPacket * const pkt, const bool verbose)
 {
-	if (!input || !output || !pkt) {
+	// input validation, most notably the input streams can sometimes
+	// be empty, causing out-of-bounds errors below
+	if (!input || !input->format_ctx || !input->format_ctx->streams
+          || !output || !pkt || pkt->stream_index < 0) {
```

Comment:
Interesting. Do you know which of the additional checks caught the problem? I'd imagine `input->format_ctx`, but it is strange that `input` would be non-NULL if `->format_ctx` is NULL. It still sounds like the caller may have a race problem if this is happening.